spring-core/src/main/java/org/springframework/core/MethodParameter.java

```diff
@@ -759,7 +765,47 @@ else if (ctor != null) {
 			}
 			return false;
 		}
-		
-	}
 
+		/**
+		 * Returns a return type of a method using Kotlin Reflection API.
+		 * Introduced to support suspending functions.
+		 */
+		@NonNull
+		static private Class<?> getReturnType(Method method) {
+		    if (isSuspend(method)) {
+		        return getSuspendReturnType(method).resolve();
+            } else {
+                return method.getReturnType();
+            }
+		}
+
+		/**
+		 * Returns a generic return type of a method using Kotlin Reflection API.
+		 * Introduced to support suspending functions.
+		 */
+		static private Type getGenericReturnType(Method method) {
+            if (isSuspend(method)) {
+                return getSuspendReturnType(method).getType();
+            } else {
+                return method.getGenericReturnType();
+            }
+		}
+
+		static private boolean isSuspend(Method method) {
+            KFunction<?> function = ReflectJvmMapping.getKotlinFunction(method);
+            return (function != null && function.isSuspend());
+        }
+
+        @NonNull
```

Comment:
Not necessary, already the default via package level `@NonNullApi`.

Reply:
Fixed.